elastic/eui
# [EuiAccordion] Split up rendering into sub-components & other cleanups

## Summary

As always, I recommend [following along by commit](https://github.com/elastic/eui/pull/7161/commits) and reading the full commit messages!

### Things this PR does

- Splits up the entirely-too-long `render()` method of `EuiAccordionClass`, and splits up styles alongside their sub-components
- Renames `.euiAccordion__iconButton` to a more descriptive `.euiAccordion__arrow` (and removes the `-isOpen` modifier classes in favor of `[aria-expanded="true"]`)
- Fixes EuiAccordion's open/close height animation to respect reduced motion media queries
- Adds a `:focus-visible` outline around EuiAccordion children on open to make it more visibly clearer that focus has been moved
- Removes an extra `<div>` wrapper around accordion children
- Removes the entirely unnecessary `tabbable()` logic I added in #7064. I realized while working on this that the [inert property](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert) (which received full cross-browser support earlier this year in spring 2023) does exactly what we want without all that extra code 🤦 

### Things this PR should not do

- Regress or break existing EuiAccordion functionality in any way
- Does not add unit tests for each newly-broken-down subcomponent (although I can do so if strongly requested) - currently counting on our existing `accordion.test.tsx` and `accordion.spec.tsx` files to regression test for us

## QA

- Go to https://eui.elastic.co/pr_7161/#/layout/accordion
- [x] Smoke test various accordions on the page and confirm they all still work as before
- `gh pr checkout 7161 && yarn storybook`
- Go to http://localhost:6006/?path=/story/euicollapsiblenavitem--edge-case-testing
- [x] Close the open accordion with multiple nested accordions and press tab, and confirm the hidden children are not tabbed to
- _Self-QA note: Tested the repro steps in #6031 and confirmed draggable accordions still do not have issues_

### General checklist

- [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) snapshots ~and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests~**
- ~[ ] Checked for **breaking changes** and labeled appropriately~ - className change is a minor breaking change, I'll handle this myself in the next Kibana upgrade


~- [ ] Checked in both **light and dark** modes~
~- [ ] Checked in **mobile**~
~- [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
